src/flamegpu/simulation/CUDAEnsemble.cu

```diff
@@ -172,6 +185,36 @@ unsigned int CUDAEnsemble::simulate(const RunPlanVector& plans) {
     // Return to device 0 (or check original device first?)
     gpuErrchk(cudaSetDevice(0));
 
+    // If there are no devices left (and mpi is not being used), we need to error as the work cannot be executed.
+#ifdef FLAMEGPU_ENABLE_MPI
+    if (devices.size() == 0 && mpi == nullptr) {
+        THROW exception::InvalidCUDAdevice("FLAMEGPU2 has not been built with an appropraite compute capability for any devices, unable to continue\n");
+    }
+#else  // FLAMEGPU_ENABLE_MPI
+    if (devices.size() == 0) {
+        THROW exception::InvalidCUDAdevice("FLAMEGPU2 has not been built with an appropraite compute capability for any devices, unable to continue\n");
+    }
+#endif  // FLAMEGPU_ENABLE_MPI
+
+#ifdef FLAMEGPU_ENABLE_MPI
+    // Once the number of devices per rank is known, we can create the actual communicator to be used during MPI, so we can warn/error as needed.
+    if (mpi != nullptr) {
```

Comment:
Redundant condition?

Reply:
Based on this. 

https://github.com/FLAMEGPU/FLAMEGPU2/blob/cdad83a2dd9c34e2ebefc3a7fbea740180e7a62e/src/flamegpu/simulation/CUDAEnsemble.cu#L66